vertx-web/src/main/java/io/vertx/ext/web/handler/impl/BodyHandlerImpl.java

```diff
@@ -160,30 +179,35 @@ public BHandler(RoutingContext context, long contentLength) {
       initBodyBuffer(contentLength);
 
       if (isMultipart || isUrlEncoded) {
-        makeUploadDir(context.vertx().fileSystem());
         context.request().setExpectMultipart(true);
```

Comment:
File uploads are just possible with multipart mode active so the flag should affect this line

Reply:
Originally I did that but when I wrote the test for "application/x-www-form-urlencoded" (`testNoUploadDirFormURLEncoded()`) I foud out that `setExpectMultipart(true)` is needed also for it because it is installing a decoder that manages both content types so, the current implementation, keeps it and just discars the "file uploads" in multimart requests, preserving any parameters.

Reply:
I'm not sure here anymore, but couldn't it be because in the `end()` method the form params are only merged if the expect multipart flag is true?

Reply:
You've a point here, thanks.